### PR TITLE
Fix #1314: Support for EC Private Keys

### LINKED ISSUE #1314: Support for EC Private Keys
Hi,

The keys I have are EC Private Keys. 
-----BEGIN EC PRIVATE KEY-----

I noticed the official java kubernetes-client added support for EC Private keys in Feb 2018. 
Any plans to add support? 

```
Caused by: java.io.IOException: Invalid DER: object is not integer
        at io.fabric8.kubernetes.client.internal.PKCS1Util$Asn1Object.getInteger(PKCS1Util.java:125)
        at io.fabric8.kubernetes.client.internal.PKCS1Util.next(PKCS1Util.java:55)
        at io.fabric8.kubernetes.client.internal.PKCS1Util.decodePKCS1(PKCS1Util.java:46)
        at io.fabric8.kubernetes.client.internal.CertUtils.createKeyStore(CertUtils.java:116)
        at io.fabric8.kubernetes.client.internal.CertUtils.createKeyStore(CertUtils.java:197)
        at io.fabric8.kubernetes.client.internal.SSLUtils.keyManagers(SSLUtils.java:128)
        at io.fabric8.kubernetes.client.internal.SSLUtils.keyManagers(SSLUtils.java:122)
        at io.fabric8.kubernetes.client.utils.HttpClientUtils.createHttpClient(HttpClientUtils.java:78)
```

### PR DESCRIPTION
**Pull request by rohanKanojia**
Fix #1314

### CODE REVIEW

**Comment by traviswinter** [comment]
@rohanKanojia - just wanted to check the status of this?  Looks like you just released 4.1.3, any chance this will be in the next? :)

**Comment by rohanKanojia** [comment]
@traviswinter: Oh yeah, thanks for reminding. Will get this in before releasing 4.1.4

**Comment by traviswinter** [comment]
No, thanks you! We just discovered that Docker Enterprise Kubernetes seems to use EC certs by default :)

File: pom.xml [hunk 1]
```diff
@@ -99,7 +99,7 @@
 
     <slf4j.version>1.7.13</slf4j.version>
     <snakeyaml.version>1.23</snakeyaml.version>
-
+    <bouncycastle.version>1.59</bouncycastle.version>
```
Comment (gastaldi) [inline pom.xml]:
```suggestion
    <bouncycastle.version>1.61</bouncycastle.version>
```

**Comment by iocanel** [comment]
I'd prefer if we split the `loadKey` method into two distinct implementations. The existing one could be the default and live the client maven module. The EC one could live in a different maven module.

The selection of impl, could be done using java.util.ServiceLoader.

This way, the client jar won't depend on bouncycastle.

**Comment by gastaldi** [comment]
The BouncyCastle dependency should be `<optional>true</optional>` and added by the clients that require this. IMHO the kubernetes-client library shouldn't enforce a specific javax.crypto library

File: platforms/karaf/features/src/main/resources/feature.xml [hunk 2]
```diff
@@ -25,6 +25,8 @@
     <bundle dependency='true'>mvn:com.fasterxml.jackson.core/jackson-databind/${jackson.bundle.version}</bundle>
     <bundle dependency='true'>mvn:com.fasterxml.jackson.dataformat/jackson-dataformat-yaml/${jackson.bundle.version}</bundle>
     <bundle dependency='true'>mvn:org.yaml/snakeyaml/${snakeyaml.version}</bundle>
+    <bundle dependency='true'>mvn:org.bouncycastle/bcprov-ext-jdk15on/${bouncycastle.version}</bundle>
```
Comment (gastaldi) [inline platforms/karaf/features/src/main/resources/feature.xml]: Karaf needs to bundle this dependency? I thought they were optional

Comment (rohanKanojia) [reply platforms/karaf/features/src/main/resources/feature.xml]: oops, I forgot to update

Comment (rohanKanojia) [reply platforms/karaf/features/src/main/resources/feature.xml]: strange, karaf tests started breaking when I removed them :confused:

**Pull request merged**